Pear/Controller/FeedbackViewController.swift

```diff
@@ -135,60 +154,116 @@ class FeedbackViewController: UIViewController {
         detailTextView.isEditable = true
         detailTextView.isHidden = true
         view.addSubview(detailTextView)
-
+        
+        dismissKeyboard()
         setupConstraints()
     }
+    
+    func dismissKeyboard() {
+           let tap = UITapGestureRecognizer(target: self, action: #selector(dismissKeyboardTouchOutside))
+           tap.cancelsTouchesInView = false
+           view.addGestureRecognizer(tap)
+    }
+        
+    @objc private func dismissKeyboardTouchOutside() {
+       view.endEditing(true)
+    }
+
+    @objc private func finishFeedback() {
+        let didMeet = responses[0] == "Yes"
+        let rating = Int(responses[1])
+        let reasonsNo = Array(responses[1...])
+        NetworkManager.sendFeedback(matchId: matchId, didMeet: didMeet, rating: rating, didMeetReason: detailTextView.text, didNotMeetReasons: reasonsNo) { [weak self] success in
+            guard let self = self else { return }
+            DispatchQueue.main.async {
+                if success {
+                    self.dismiss(animated: true, completion: nil)
+                } else {
+                    self.present(UIAlertController.getStandardErrortAlert(), animated: true)
+                }
+            }
+        }
+    }
+    
+    @objc private func quitFeedback() {
+        NetworkManager.quitFeedback() { [weak self] success in
+            guard let self = self else { return }
+            DispatchQueue.main.async {
+                if success {
+                    self.dismiss(animated: true, completion: nil)
+                } else {
+                    self.present(UIAlertController.getStandardErrortAlert(), animated: true)
+                }
+            }
+        }
+    }
+    
+    @objc private func goBack() {
+        backButton.isHidden = true
+        finishButton.isHidden = true
+        finishButton.isEnabled = false
+        responses = []
+        detailTextView.text = ""
+        questionLabel.text = "Did you meet your last Pear?"
+        feedbackOptions = initialFeedbackOptions
+        feedbackStatus = .initial
+        answerCollectionView.allowsMultipleSelection = false
+        lowRatingLabel.isHidden = true
+        highRatingLabel.isHidden = true
+        askDetailLabel.isHidden = true
+        detailTextView.isHidden = true
+        answerCollectionView.snp.remakeConstraints { make in
+            make.leading.trailing.equalToSuperview().inset(36)
+            make.top.equalTo(questionLabel.snp.bottom).offset(10)
+            make.height.equalTo(200)
+        }
+        DispatchQueue.main.async {
+            self.answerCollectionView.reloadData()
+        }
+    }
```

Comment:
this whole function seems a little hard to read. Maybe add a spacing between each component that needs to be configured.